client/apps/edit/components/content/sections/image_collection/components/controls.jsx

```diff
@@ -1,186 +1,197 @@
+import PropTypes from 'prop-types'
 import React, { Component } from 'react'
-import { clone, defer } from 'lodash'
+import { clone } from 'lodash'
+import { connect } from 'react-redux'
+import { data as sd } from 'sharify'
+import { Row, Col } from 'react-styled-flexboxgrid'
 import Artwork from '/client/models/artwork.coffee'
-import Autocomplete from '/client/components/autocomplete/index.coffee'
-import FileInput from '/client/components/file_input/index.jsx'
-import SectionControls from '../../../section_controls/index.jsx'
-import UrlArtworkInput from './url_artwork_input.coffee'
-import PropTypes from 'prop-types'
-const sd = require('sharify').data
+import FileInput from '/client/components/file_input'
+import SectionControls from '../../../section_controls'
+import { logError } from 'client/actions/editActions'
+import { Autocomplete } from '/client/components/autocomplete2'
+import { InputArtworkUrl } from './input_artwork_url'
 
-export default class Controls extends Component {
+export class ImageCollectionControls extends Component {
   static propTypes = {
-    article: PropTypes.object,
-    channel: PropTypes.object,
-    images: PropTypes.array,
+    articleLayout: PropTypes.string.isRequired,
+    channel: PropTypes.object.isRequired,
     isHero: PropTypes.bool,
-    onChange: PropTypes.func,
-    section: PropTypes.object,
+    logErrorAction: PropTypes.func,
+    section: PropTypes.object.isRequired,
     setProgress: PropTypes.func
   }
 
-  componentDidMount () {
-    this.setupAutocomplete()
-  }
+  componentWillUnmount = () => {
+    const { section } = this.props
 
-  componentWillUnmount () {
-    this.autocomplete.remove()
+    if (!section.get('images').length) {
+      section.destroy()
+    }
   }
 
-  addArtworkFromUrl = (images) => {
-    this.props.section.set('images', images)
-    this.props.onChange()
-  }
-
-  setupAutocomplete () {
-    const $el = $(this.refs.autocomplete)
-    this.autocomplete = new Autocomplete({
-      url: `${sd.ARTSY_URL}/api/search?q=%QUERY`,
-      el: $el,
-      filter (res) {
-        const vals = []
-        for (let r of Array.from(res._embedded.results)) {
-          if ((r.type != null ? r.type.toLowerCase() : undefined) === 'artwork') {
-            const id = r._links.self.href.substr(r._links.self.href.lastIndexOf('/') + 1)
-            vals.push({
-              id,
-              value: r.title,
-              thumbnail: (r._links.thumbnail != null ? r._links.thumbnail.href : null)
-            })
-          }
-        }
-        return vals
-      },
-      templates: {
-        suggestion (data) {
-          return `<div class='autocomplete-suggestion' style='background-image: url(${data.thumbnail})'></div>${data.value}`
+  filterAutocomplete = (items) => {
+    return items._embedded.results.map((item) => {
+      const { type } = item
+
+      if (type === 'artwork') {
```

Comment:
It looks like the old code has a `r.type.toLowerCase` check. Is it verified that Artwork types will always be lowercase?

Reply:
Good catch -- will add this back